src/main/java/com/gluonhq/substrate/ProjectConfiguration.java

```diff
@@ -0,0 +1,246 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.substrate;
+
+import com.gluonhq.substrate.model.IosSigningConfiguration;
+import com.gluonhq.substrate.model.Triplet;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This class contains all public configuration info about the current project (not about the current OS/Arch/vendor etc)
+ * Instances of this class give hints to where real values should be contained from.
+ */
+public class ProjectConfiguration {
+
+    private String graalPath;
+
+    private String javafxStaticSdkVersion;
+    private String javaStaticSdkVersion;
+
+    private boolean usePrismSW = false;
+    private boolean verbose = false;
+
+    private Triplet targetTriplet;
+    private Triplet hostTriplet = Triplet.fromCurrentOS();
+
+    private List<String> bundlesList = Collections.emptyList();
+    private List<String> resourcesList = Collections.emptyList();
+    private List<String> reflectionList = Collections.emptyList();
+    private List<String> jniList = Collections.emptyList();
+
+    private String appName;
+    private String mainClassName;
+
+    private IosSigningConfiguration iosSigningConfiguration = new IosSigningConfiguration();
+
+    public ProjectConfiguration( String mainClassName ) {
+        this.mainClassName = Objects.requireNonNull(mainClassName, "Main class name is required")
+                               .contains("/") ?
+                                  mainClassName.substring( mainClassName.indexOf("/") + 1) : mainClassName;
+
+    }
+
+    public Path getGraalPath() {
+        return Path.of( Objects.requireNonNull(this.graalPath, "GraalVM Path is not defined"));
+    }
+
+    public void setGraalPath(String path) {
+        this.graalPath = path;
+    }
+
+
+    /**
+     * Sets the Java static SDK version
+     * @param javaStaticSdkVersion the Java static SDK version
+     */
+    public void setJavaStaticSdkVersion(String javaStaticSdkVersion) {
+        this.javaStaticSdkVersion = javaStaticSdkVersion;
+    }
+
+    public String getJavaStaticSdkVersion() {
+        return Objects.requireNonNull( this.javaStaticSdkVersion, "Java Static SDK version is required" );
```

Comment:
This causes tests to fail, and I think behavior should be as follows:
* if the developer somehow specified a "javastaticsdkversion" (which is highly discouraged, I think we don't even want to expose this), that "version" is returned. 
* else, a Constants value should be returned.

Reply:
The logic should probably be in the PrivateProjectConfiguration. The ProjectConfiguration itself is simply a value holder for e.g. plugins to advise on configuration

Reply:
Fixed